js/ui/map.js

```diff
@@ -734,6 +734,23 @@ class Map extends Camera {
     }
 
     /**
+     * Returns a Boolean indicating whether the source is loaded.
+     *
+     * @param {string} id The ID of the source to be checked.
+     * @returns {boolean} A Boolean indicating whether the source is loaded.
+     */
+    isSourceLoaded(id) {
+        const source = this.style.sourceCaches[id];
```

Comment:
Note that it's possible for `Map` to be constructed without a style, which would cause this line to blow up.  Something like `const source = this.style && this.style.sourceCaches[id];` would work.